x-pack/plugins/security/public/management/users/users_grid/users_grid_page.tsx

```diff
@@ -112,7 +112,7 @@ export class UsersGridPage extends Component<Props, State> {
         render: (username: string) => (
           <EuiLink
             data-test-subj="userRowUserName"
-            {...reactRouterNavigate(this.props.history, `/edit/${username}`)}
+            {...reactRouterNavigate(this.props.history, `/edit/${encodeURIComponent(username)}`)}
```

Comment:
__question:__ how do you feel about doing the same for the role name in Roles Grid? I see we don't do encoding there. I know Kibana UI doesn't allow creating roles with such names, but ES API does. At least I managed to create role with such a name using Kibana DevTools and it broke our UI:
```http
POST /_security/role/my_admin%25role
{
  "cluster": ["all"],
  "indices": [
    {
      "names": [ "index1", "index2" ],
      "privileges": ["all"]
    }
  ],
  "applications": [
    {
      "application": "myapp",
      "privileges": [ "admin", "read" ],
      "resources": [ "*" ]
    }
  ]
}
```

Reply:
Great idea, will do!

Reply:
Done in [`13be3a0` (#81562)](https://github.com/elastic/kibana/pull/81562/commits/13be3a09e1400245424c155dbe5890e9924446d8)